snorkel/learning/gen_learning.py

```diff
@@ -184,55 +188,153 @@ def __init__(self, class_prior=False, lf_prior=False, lf_propensity=False, lf_cl
         self.rng = random.Random()
         self.rng.seed(seed)
 
-    # These names of factor types are for the convenience of several methods that perform the same operations over
-    # multiple types, but this class's behavior is not fully specified here. Other methods, such as marginals(),
-    # as well as maps defined within methods, require manual adjustments to implement changes.
+    # These names of factor types are for the convenience of several methods 
+    # that perform the same operations over multiple types, but this class's 
+    # behavior is not fully specified here. Other methods, such as marginals(),
+    # as well as maps defined within methods, require manual adjustments to 
+    # implement changes.
     #
-    # These names are also used by other related classes, such as GenerativeModelParameters
+    # These names are also used by other related classes, such as 
+    # GenerativeModelParameters
     optional_names = ('lf_prior', 'lf_propensity', 'lf_class_propensity')
-    dep_names = ('dep_similar', 'dep_fixing', 'dep_reinforcing', 'dep_exclusive')
-
-    def train(self, L, y=None, deps=(), init_acc = 1.0, init_deps=1.0, init_class_prior=-1.0, epochs=10, step_size=None, decay=0.99, reg_param=0.1, reg_type=2, verbose=False,
-              truncation=10, burn_in=5, timer=None):
+    dep_names = (
+        'dep_similar', 'dep_fixing', 'dep_reinforcing', 'dep_exclusive'
+    )
+
+    def train(self, L, deps=(), LF_priors=None, LF_prior_default=0.7, 
+        labels=None, label_prior=0.95, init_deps=1.0, 
+        init_class_prior=-1.0, epochs=100, step_size=None, decay=0.99, 
```

Comment:
I believe one issue (that @bryan-he pointed out) is that the step size will still be quite large after only 10 epochs.  This could be addressed by dynamically setting the step size decay factor... but either way thinking (from tests even) was that 10 is too small?  @bryan-he any thoughts?